### PR TITLE
Task runner

### PR DESCRIPTION
**Pull request by AlliumApotheosis**
stuff i've been throwing around, very WIP
@benbp

### CODE REVIEW

**Comment by benbp** [comment]
want to rebase out those other commits from zach/joel?

File: spec/lib/task-runner-spec.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    });
+
+    it("should instantiate and run checked out task", function(done) {
+        store.checkoutTask.resolves(taskDef);
+        runner.inputStream.onNext(taskAndGraphId);
+
+        setImmediate(function() {
+            expect(Task.create).to.have.been.calledOnce;
+            expect(stubbedTask.run).to.have.been.calledOnce;
+            done();
```
Comment (benbp) [inline spec/lib/task-runner-spec.js:58]: Try/Catch all these done blocks :)

Comment (benbp) [inline spec/lib/task-runner-spec.js:58]: I'm one to talk, I don't even have tests yet.

File: lib/task-runner.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    TaskRunner.prototype.start = function() {
+        this.initializePipeline();
+
+        return taskMessenger.subscribeRun(this.domain, this.inputStream.onNext);
+    };
+
+    TaskRunner.prototype.initializePipeline = function() {
+        this.pipeline = this.inputStream.flatMap(function(taskAndGraphId) {
+            return store.checkoutTask(taskAndGraphId.taskId);
```
Comment (benbp) [inline lib/task-runner.js:54]: Don't we need the graphId for this anyways?

File: spec/lib/task-runner-spec.js (hunk 1 above)
Comment (AlliumApotheosis) [inline spec/lib/task-runner-spec.js:58]: my tests aren't dooooonne

**Comment by benbp** [comment]
Some comments for the future. I'll merge though since it's a dev branch :P

**Pull request merged**